commit caf4df71d2bbb26944c6afbe39912ee0a1c2537a
Author: Niels Roest <niels@directfb.org>
Date:   Mon Nov 23 14:03:25 2009 +0100

    skirmish: fix for transferring transferred skirmishes
    
    This behaviour was initially asserted on, but it is now
    simply disallowed to transfer transferred skirmishes,
    since the ASSERT will hit in some cases in normal circumstances.

diff --git a/linux/drivers/char/fusion/skirmish.c b/linux/drivers/char/fusion/skirmish.c
index b9c22a2..f1f48f9 100644
--- a/linux/drivers/char/fusion/skirmish.c
+++ b/linux/drivers/char/fusion/skirmish.c
@@ -678,8 +678,7 @@ fusion_skirmish_transfer_all(FusionDev * dev,
 
 		down(&skirmish->entry.lock);
 
-		if (skirmish->lock_pid == from_pid) {
-			FUSION_ASSERT(skirmish->transfer_to == 0);
+		if ( (skirmish->lock_pid == from_pid) && (skirmish->transfer_to == 0) ) {
 			FUSION_ASSERT(skirmish->transfer_from == 0);
 			FUSION_ASSERT(skirmish->transfer_from_pid == 0);
 			FUSION_ASSERT(skirmish->transfer_count == 0);
